### ISSUE TITLE
azurerm_sql_server: name must be lower case

### ISSUE THREAD

**Issue by StefanSchoof** [terraform-providers/terraform-provider-azurerm#314]
### Terraform Version
0.10.3

### Affected Resource(s)
- azurerm_sql_server

### Terraform Configuration Files
```hcl
resource "azurerm_resource_group" "production" {
  name     = "terraformTest"
  location = "West Europe"
}

resource "azurerm_sql_server" "test" {
  name="MysqlServerasdsadasdasdas"
  resource_group_name="${azurerm_resource_group.production.name}"
  location="${azurerm_resource_group.production.location}"
  version="12.0"
  administrator_login="xxx"
  administrator_login_password="xxx"
}
```

### Expected Behavior
If try to use a name with Uppercase chars in the name, the behavior should be better.
The documentation or/and the system says that the name has to be lowercase or convert the name to lowercase.

### Actual Behavior
`terraform apply` fails with a `StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="Unknown" Message="Unknown service error"`

### Steps to Reproduce
1. `terraform apply`

### References
- GH-241
- https://www.terraform.io/docs/providers/azurerm/r/sql_server.html

**Comment by StefanSchoof** [comment]
Add link to documentation

**Comment by tombuildsstuff** [comment]
Closed via #323
